From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Fri, 14 Apr 2023 17:05:19 +0200
Subject: [PATCH] Create eula.txt before bootstrap

Moves the `eula.txt` file creation to before the server bootstrap (which includes, amongst others, `Bootstrap.bootStrap`).

This makes sure the server owner first can deal with accepting the EULA before dealing with anything going wrong during bootstrapping when the server is first started (for example, because there are Fiddle packs with errors present already, or because a plugin causes an error during bootstrapping).

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/main/java/net/minecraft/server/Main.java b/src/main/java/net/minecraft/server/Main.java
index 37cdb4be04251e14118e71d939a6d73ad31c42ea..66803db4244b6eb28d51fbc11d0a2c26a6171021 100644
--- a/src/main/java/net/minecraft/server/Main.java
+++ b/src/main/java/net/minecraft/server/Main.java
@@ -123,6 +123,11 @@ public class Main {
                 JvmProfiler.INSTANCE.start(Environment.SERVER);
             }
 
+            // Fiddle start - create eula.txt before bootstrap
+            Path path2 = Paths.get("eula.txt");
+            Eula eula = new Eula(path2);
+            // Fiddle end - create eula.txt before bootstrap
+
             // Fiddle start - Fiddle configuration - load Fiddle global configuration before bootstrap
             var fiddleConfigurations = org.fiddlemc.fiddle.configuration.FiddleConfigurations.get(optionset);
             fiddleConfigurations.initializeGlobalConfiguration();
@@ -136,8 +141,6 @@ public class Main {
             DedicatedServerSettings dedicatedserversettings = new DedicatedServerSettings(optionset); // CraftBukkit - CLI argument support
 
             dedicatedserversettings.forceSave();
-            Path path2 = Paths.get("eula.txt");
-            Eula eula = new Eula(path2);
             // Paper start - load config files for access below if needed
             org.bukkit.configuration.file.YamlConfiguration bukkitConfiguration = io.papermc.paper.configuration.PaperConfigurations.loadLegacyConfigFile((File) optionset.valueOf("bukkit-settings"));
             org.bukkit.configuration.file.YamlConfiguration spigotConfiguration = io.papermc.paper.configuration.PaperConfigurations.loadLegacyConfigFile((File) optionset.valueOf("spigot-settings"));
